### PR TITLE
Add support for Spark 2.4.0-hadoop2.8-scala2.12

### PR DESCRIPTION
**Pull request by shimamoto**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
```
Comment (GezimSejdiu) [inline build.sh]: Could you please, change `build python-template template/scala` to `build scala-template template/scala` at [line 21](https://github.com/big-data-europe/docker-spark/blob/1c9faf52b82c2fbbb526ccdc3fc0cdc97ce412f2/build.sh#L21)?

File: template/scala/build.sbt [hunk 2]
```diff
@@ -1,2 +1,2 @@
 sparkVersion := "2.4.0"
-scalaVersion := "2.11.12"
+scalaVersion := "2.12.7"
```
Comment (GezimSejdiu) [inline template/scala/build.sbt]: Seems that [sbt-spark-package](https://github.com/databricks/sbt-spark-package) (see [#41](https://github.com/databricks/sbt-spark-package/pull/41)) does not support Scala 2.12 yet. When we use the current version it is not able to find that dependency and gives us : 
```sh
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: org.spark-packages#sbt-spark-package;0.2.6: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]
[warn]  Note: Some unresolved dependencies have extra attributes.  Check that these dependencies exist with the requested attributes.
[warn]          org.spark-packages:sbt-spark-package:0.2.6 (scalaVersion=2.12, sbtVersion=1.0)
[warn]
[warn]  Note: Unresolved dependencies path:
[warn]          org.spark-packages:sbt-spark-package:0.2.6 (scalaVersion=2.12, sbtVersion=1.0) (/app/project/plugins.sbt#L2-3)
[warn]            +- default:app-build:0.1.0-SNAPSHOT (scalaVersion=2.12, sbtVersion=1.0)
[error] sbt.librarymanagement.ResolveException: unresolved dependency: org.spark-packages#sbt-spark-package;0.2.6: not found
```
One workaround would be to use the same dependency resolver as suggested on [#41](https://github.com/databricks/sbt-spark-package/pull/41) and the build pass successfuly. Just replace the content on [plugins.sbt](https://github.com/big-data-europe/docker-spark/blob/1c9faf52b82c2fbbb526ccdc3fc0cdc97ce412f2/template/scala/plugins.sbt), with these two lines 
```sh
resolvers += Resolver.bintrayIvyRepo("s22s", "sbt-plugins")
addSbtPlugin("org.spark-packages" % "sbt-spark-package" % "0.2.7-astraea.1")
```
PS: When they release a new version of sbt-spark-package which reslove this, we may need to update the resolvers then.

Comment (GezimSejdiu) [reply template/scala/build.sbt]: Or maybe it is just related to the sbt version older than 1.x?, mine is at `1.2.6`, but this workaround is working for me. Give it a try and let me know.

File: build.sh (hunk 1 above)
Comment (shimamoto) [reply build.sh]: It looks like this line is commented out on master branch. Would you mind if I change it on this branch?

Comment (GezimSejdiu) [reply build.sh]: Sure, feel free to change it here!

File: template/scala/build.sbt (hunk 2 above)
Comment (GezimSejdiu) [reply template/scala/build.sbt]: Right, let's give it a try and see how it goes. Use sbt-assembly instead!

**Comment by shimamoto** [comment]
Fixed it!

**Pull request merged**